internal/services/netapp/netapp_volume_resource.go

```diff
@@ -1121,6 +1121,10 @@ func flattenNetAppVolumeDataProtectionSnapshotPolicy(input *volumes.VolumeProper
 		return []interface{}{}
 	}
 
+	if input.Snapshot != nil && *input.Snapshot.SnapshotPolicyId == "" {
```

Comment:
We can consolidate into this [line.](https://github.com/hashicorp/terraform-provider-azurerm/pull/18348/files#diff-d193f61e6316f1a73de8bfe688c1abc0b068bf06213b8ed9fc824238b44a08f4R1120)

```
if input == nil || input.Snapshot == nil || *input.Snapshot.SnapshotPolicyId == ""
```

Reply:
Hello @mbfrahry, thanks for the feedback!
I changed the corresponding code and added an new commit.